lib/Service/PageService.php

```diff
@@ -942,7 +942,7 @@ public function matchBacklinks(PageInfo $pageInfo, string $content): bool {
 
 		$protocol = 'https?:\/\/';
 		$trustedDomainConfig = (array)$this->config->getSystemValue('trusted_domains', []);
-		$trustedDomains = !empty($trustedDomainConfig) ? '(' . implode('|', $trustedDomainConfig) . ')' : 'localhost';
+		$trustedDomains = !($trustedDomainConfig === []) ? '(' . implode('|', $trustedDomainConfig) . ')' : 'localhost';
```

Comment:
You could avoid the `!` in the condition here - either by using the non-identity operator or by changing the order of the ternary operator.
```suggestion
		$trustedDomains = ($trustedDomainConfig !== []) ? '(' . implode('|', $trustedDomainConfig) . ')' : 'localhost';
```

Reply:
Uh, nice catch :ok_hand: